File: src/main/java/lsystem/LSystem.java
package lsystem;

public abstract class AbstractLSystem implements LSystem {

    private String _axiom;
    private Rules _rules;

    public AbstractLSystem(String axiom, Rules rules) {
        _axiom = axiom;
    }

    @Override
    public String computeGeneration(int generation) {

    }

}

File: script/LSystem.js
class LSystem {

	constructor(axiom, rules) {
		this.generations = [axiom];
		this.rules = rules;
	}
	
	wordAtGeneration(gen) {
		if (this.generations.length > gen) {
			return this.generations[gen];
		}
		var word = wordAtGeneration(gen - 1);
		var newWord = "";
		for (const symbol of word) {
			newWord += this.rules.getMutation(symbol);
		}
		this.generations[gen] = newWord;
		return newWord;
	}

}

